packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -641,10 +643,11 @@ export class TeamsAppSolution implements Solution {
 
       // Only Azure project requires this confirm dialog
       const username = (azureToken as any).username ? (azureToken as any).username : "";
-      const subscriptionInfo = await ctx.azureAccountProvider?.getSelectedSubscription();
+      const subscriptionId = ctx.envInfo.profile.get(GLOBAL_CONFIG)?.get(SUBSCRIPTION_ID) as string;
```

Comment:
Where to set the subscription info before to reading it?

Reply:
Solution has already put subscription info into `ctx.envInfo.profile` before provision (similar to what we do with the resource group info). So this PR is just a refactor. Plugins can read directly from ctx.envInfo.profile because solution has already put subs in it.

Reply:
I see. Thanks for the explanation.